### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by veeding**
complete import task and pass all test

### CODE REVIEW

File: server-v2/api/studio/restapi/import.api [hunk 1]
```diff
@@ -0,0 +1,243 @@
+syntax = "v1"
```
Comment (veezhang) [inline server-v2/api/studio/restapi/import.api:1]: Can you adjust it according to this pr, look at the comments?
https://github.com/vesoft-inc/nebula-cloud/pull/808

File: server-v2/api/studio/internal/logic/file/filedestroylogic.go [hunk 2]
```diff
@@ -1,8 +1,8 @@
 package file
 
 import (
-	"context"
 	"github.com/vesoft-inc/nebula-studio/server/api/studio/internal/service"
+	"net/http"
```
Comment (veezhang) [inline server-v2/api/studio/internal/logic/file/filedestroylogic.go]: Forget to `make check`?

File: server-v2/api/studio/internal/logic/file/filedestroylogic.go [hunk 3]
```diff
@@ -12,18 +12,18 @@ import (
 
 type FileDestroyLogic struct {
 	logx.Logger
-	ctx    context.Context
+	r      *http.Request
```
Comment (veezhang) [inline server-v2/api/studio/internal/logic/file/filedestroylogic.go]: Maybe this pr https://github.com/vesoft-inc/go-pkg/pull/18 can help you.

File: server-v2/api/studio/internal/service/file.go [hunk 4]
```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/saintfish/chardet"
 	"github.com/vesoft-inc/nebula-studio/server/api/studio/internal/svc"
 	"github.com/vesoft-inc/nebula-studio/server/api/studio/internal/types"
+	Config "github.com/vesoft-inc/nebula-studio/server/api/studio/pkg/config"
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/file.go]: The alias of pkg usually lowercase. And this pkg will be removed.

File: server-v2/api/studio/internal/service/import.go [hunk 5]
```diff
@@ -0,0 +1,317 @@
+package service
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	importconfig "github.com/vesoft-inc/nebula-importer/pkg/config"
+	importerErrors "github.com/vesoft-inc/nebula-importer/pkg/errors"
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: The alias of pkg usually lowercase.

File: server-v2/api/studio/internal/service/import.go [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+		Logger: logx.WithContext(ctx),
+		ctx:    ctx,
+		svcCtx: svcCtx,
+	}
+}
+
+func (i *importService) CreateImportTask(req *types.CreateImportTaskRequest) (*types.CreateImportTaskData, error) {
+	jsons, err := json.Marshal(req.Config)
+	if err != nil {
+		return nil, errors.New("importDataParams get fail")
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: How about use `ecode.WithCode(ecode.ErrParam, nil)` ?

File: server-v2/api/studio/internal/service/import.go [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+	}
+
+	conf := importconfig.YAMLConfig{}
+	err = json.Unmarshal(jsons, &conf)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = validClientParams(&conf); err != nil {
+		err = importerErrors.Wrap(importerErrors.InvalidConfigPathOrFormat, err)
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: Why? Does the web care about it?

File: server-v2/api/studio/internal/service/import.go [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+			zap.L().Warn("read taskId file error", zap.Error(err))
+			return nil, err
+		}
+	}
+	taskIdJSON[taskID] = true
+	bytes, err := json.Marshal(taskIdJSON)
+	if err != nil {
+		zap.L().Warn("read taskId file error", zap.Error(err))
+	}
+	err = ioutil.WriteFile(Config.Cfg.Web.TaskIdPath, bytes, 777)
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: Why not use db to save those?

File: server-v2/api/studio/internal/service/import.go [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+		return nil, errors.New("invalid Id")
+	}
+	configPath := filepath.Join(Config.Cfg.Web.TasksDir, req.Id, "config.yaml")
+	body, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return nil, err
+	}
+
+	data := &types.DownloadConfigsData{}
+	data.Data = string(body[:])
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: Why not use `string(body)`?

File: server-v2/api/studio/internal/service/import.go [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
+
+	return data, nil
+}
+
+func (i *importService) DownloadLogs(req *types.DownloadLogsRequest) (*types.DownloadLogsData, error) {
+	id := req.Id
+	if id == "" {
+		return nil, errors.New("id parse failed")
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: How about `ecode.ErrParam`?

File: server-v2/api/studio/internal/service/import.go [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+	}
+	filename := req.Name
+	path := ""
+	if filename == "import.log" {
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: I don't think this is a good way to differentiate between normal logs and error logs. I remember you talking about adding a category. What do you think?

File: server-v2/api/studio/internal/service/import.go [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+	errLogDir := filepath.Join(Config.Cfg.Web.TasksDir, id, "err")
+	fileInfos, err := ioutil.ReadDir(errLogDir)
+	if err != nil {
+		return nil, err
+	}
+
+	data := &types.GetImportTaskLogNamesData{
+		Names: []string{},
+	}
+	data.Names = append(data.Names, "import.log")
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: I have seen `import.log` many times, how about defining it as a constant?

File: server-v2/api/studio/internal/service/import.go [hunk 13]
```diff
@@ -0,0 +1,317 @@
...
+		if fn.CSV.Delimiter == nil || *fn.CSV.Delimiter == "" {
+			delimiter := ","
+			fn.CSV.Delimiter = &delimiter
+		}
+	}
+
+	return nil
+}
+
+func readFile(path string, offset int64, limit int64) ([]string, error) {
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: How about use `readFileLines` or similar.

File: server-v2/api/studio/internal/service/importer/importer.go [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+	taskDir := filepath.Join(Config.Cfg.Web.TasksDir, taskId)
+	return taskDir, nil
+}
+
+func CreateConfigFile(dir string, config importconfig.YAMLConfig) error {
+	fileName := "config.yaml"
+	err := utils.CreateDir(dir)
+	if err != nil {
+		return err
+	}
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/importer/importer.go]: refer to uber go style.
```
if err := utils.CreateDir(dir); err != nil {
	return err
}
```

File: server-v2/api/studio/internal/service/import.go [hunk 15]
```diff
@@ -0,0 +1,334 @@
...
+	}, nil
+}
+
+func (i *importService) StopImportTask(req *types.StopImportTaskRequest) error {
+	return importer.StopImportTask(req.Id, req.Address+":"+req.Port, req.Username)
+}
+
+func (i *importService) DownloadConfig(req *types.DownloadConfigsRequest) error {
+	if req.Id == "" {
+		return errors.New("invalid Id")
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: And you can use set tag for request param.

File: server-v2/api/studio/internal/service/import.go [hunk 16]
```diff
@@ -0,0 +1,334 @@
...
+	filename := req.Name
+	path := ""
+	if filename == "import.log" {
+		path = filepath.Join(Config.Cfg.Web.TasksDir, id, filename)
+	} else {
+		path = filepath.Join(Config.Cfg.Web.TasksDir, id, "err", filename)
+	}
+
+	fmt.Println("------------------------")
+	fmt.Println("test")
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: Please remove the debug code.

File: server-v2/api/studio/internal/service/import.go [hunk 17]
```diff
@@ -0,0 +1,334 @@
...
+
+func validClientParams(conf *importconfig.YAMLConfig) error {
+	if conf.NebulaClientSettings.Connection == nil ||
+		conf.NebulaClientSettings.Connection.Address == nil ||
+		*conf.NebulaClientSettings.Connection.Address == "" ||
+		conf.NebulaClientSettings.Connection.User == nil ||
+		*conf.NebulaClientSettings.Connection.User == "" ||
+		conf.NebulaClientSettings.Space == nil ||
+		*conf.NebulaClientSettings.Space == "" {
+		return errors.New("client params is wrong")
```
Comment (veezhang) [inline server-v2/api/studio/internal/service/import.go]: The web can not get the `client params is wrong`, is this what you expect?

File: server-v2/api/studio/pkg/auth/authorize.go [hunk 18]
```diff
@@ -39,117 +47,158 @@ func CreateToken(authData *AuthData, config *config.Config) (string, error) {
...
-	}, func(token *jwt.Token) (interface{}, error) {
-		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
-			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
-		}
-		return []byte(secret), nil
-	})
-	if err != nil {
-		return err
-	}
+func AuthMiddlewareWithCtx(svcCtx *svc.ServiceContext) rest.Middleware {
```
Comment (veezhang) [inline server-v2/api/studio/pkg/auth/authorize.go]: Why is there authentication related logic in this pr?

File: server-v2/api/studio/pkg/config/config.go [hunk 19]
```diff
@@ -1,4 +1,4 @@
-package config
+package Config
```
Comment (veezhang) [inline server-v2/api/studio/pkg/config/config.go]: Why?

File: server-v2/api/studio/pkg/config/config.go [hunk 20]
```diff
@@ -21,8 +21,6 @@ type (
 		UploadDir        string `yaml:"upload_dir"`
 		TasksDir         string `yaml:"tasks_dir"`
 		SqlitedbFilePath string `yaml:"sqlitedb_file_path"`
```
Comment (veezhang) [inline server-v2/api/studio/pkg/config/config.go]: How about use camel case?

File: server-v2/api/studio/pkg/ecode/codes.go [hunk 21]
```diff
@@ -46,6 +47,11 @@ func WithUnauthorized(err error, formatWithArgs ...interface{}) error {
 	return WithCode(ErrUnauthorized, err, formatWithArgs...)
 }
 
+func WithSessionMessage(err error, formatWithArgs ...interface{}) error {
+	ErrSessionWithMessage := newErrCode(CCUnauthorized, PlatformCode, 1, fmt.Sprintf("ErrSession::%s", err.Error()))
+	return WithCode(ErrSessionWithMessage, err, formatWithArgs...)
+}
```
Comment (veezhang) [inline server-v2/api/studio/pkg/ecode/codes.go]: Only very general ones are put here, this is not recommended to be put here, this file should not be easily changed.

You just need to define the new ErrCode after line 12 `// Define you error code here`.

File: server-v2/api/studio/pkg/utils/http.go [hunk 22]
```diff
@@ -1,128 +1,56 @@
...
-	"webm":    "video/webm",
-	"flv":     "video/x-flv",
-	"m4v":     "video/x-m4v",
-	"mng":     "video/x-mng",
-	"asx":     "video/x-ms-asf",
-	"asf":     "video/x-ms-asf",
-	"wmv":     "video/x-ms-wmv",
-	"avi":     "video/x-msvideo",
+// dynamicly add query params to the request
+func AddQueryParams(r *http.Request, params map[string]string) {
```
Comment (veezhang) [inline server-v2/api/studio/pkg/utils/http.go]: Modified the original query parameter, it seems strange, is there a better way?

File: server-v2/api/studio/restapi/file.api [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+	group: file
+)
+
+service studio-api {
+	@doc "Upload File"
+	@handler FileUpload
+	post /api/file/upload
+	@doc "delete file"
+	@handler FileDestroy
+	delete /api/file/:name(FileDestroyRequest)
```
Comment (veezhang) [inline server-v2/api/studio/restapi/file.api]: How about `/api/files/:name` ?

File: server-v2/api/studio/restapi/file.api [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+	@doc "preview file"
+	@handler FilesIndex
+	get /api/file returns(FilesIndexData)
```
Comment (veezhang) [inline server-v2/api/studio/restapi/file.api]: How about '/api/files`?

File: server-v2/api/studio/restapi/file.api [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+@server(
+	group: file
+)
+
+service studio-api {
+	@doc "Upload File"
+	@handler FileUpload
+	post /api/file/upload
```
Comment (veezhang) [inline server-v2/api/studio/restapi/file.api]: How about `/api/files` ?

**Comment by veezhang** [comment]
please resolve conflicts.

File: server-v2/api/studio/internal/service/import.go (hunk 11 above)
Comment (veeding) [reply server-v2/api/studio/internal/service/import.go]: I think is ok. Normal logs name is specified and error logs is in "`err` category.

File: server-v2/api/studio/internal/service/import.go (hunk 8 above)
Comment (veeding) [reply server-v2/api/studio/internal/service/import.go]: db have save those.

File: server-v2/api/studio/restapi/import.api [hunk 26]
```diff
@@ -0,0 +1,235 @@
...
+	get /api/import-tasks/:id/task-log-names(GetImportTaskLogNamesRequest) returns(GetImportTaskLogNamesData)
+	@doc "Delete Import Task"
+	@handler DeleteImportTask
+	delete /api/import-tasks/:id(DeleteImportTaskRequest)
+	@doc "Stop Import Task"
+	@handler StopImportTask
+	get /api/import-tasks/stop/:id(StopImportTaskRequest)
+	@doc "Download logs"
+	@handler DownloadLogs
+	get /api/import-tasks/download/logs/:id(DownloadLogsRequest)
```
Comment (veezhang) [inline server-v2/api/studio/restapi/import.api]: How about `/api/import-tasks/:id/download-logs` ?

File: server-v2/api/studio/restapi/import.api [hunk 27]
```diff
@@ -0,0 +1,235 @@
...
+	@doc "Download Config"
+	@handler DownloadConfig
+	get /api/import-tasks/download/config/:id(DownloadConfigsRequest)
```
Comment (veezhang) [inline server-v2/api/studio/restapi/import.api]: How about `/api/import-tasks/:id/download-config`?

File: server-v2/api/studio/studio.go [hunk 28]
```diff
@@ -38,6 +54,25 @@ func main() {
 
 	// global middleware
 	server.Use(auth.AuthMiddlewareWithCtx(svcCtx))
+	server.Use(rest.ToMiddleware(middleware.ReserveRequest(middleware.ReserveRequestConfig{
+		Skipper: func(r *http.Request) bool {
+			if strings.HasPrefix(r.URL.Path, "/api/files") {
+				return false
+			}
+			if strings.HasPrefix(r.URL.Path, "/api/import-tasks/download") {
```
Comment (veezhang) [inline server-v2/api/studio/studio.go]: There are several places with this logic,  can you encapsulated into a function?

File: server-v2/api/studio/internal/common/azurewebhook.go [hunk 29]
```diff
@@ -0,0 +1,29 @@
+package common
```
Comment (veezhang) [inline server-v2/api/studio/internal/common/azurewebhook.go]: Please use a more appropriate file name.

**Review by veezhang — APPROVED** [review]
LGTM

**Pull request merged**